scipy/stats/_stats_py.py

```diff
@@ -100,13 +100,20 @@ def _contains_nan(a, nan_policy='propagate'):
         try:
             contains_nan = np.any(np.isnan(a))
         except TypeError:
-            # Don't know what to do. Fall back to omitting nan values and
-            # issue a warning.
-            contains_nan = False
-            nan_policy = 'omit'
-            warnings.warn("The input array could not be properly "
-                          "checked for nan values. nan values "
-                          "will be ignored.", RuntimeWarning)
+            try:
+                # This can happen when attempting to check nan with np.isnan
+                # for string array (e.g. as in the function `rankdata`).
+                a = np.array(a)
```

Comment:
```suggestion
                a = np.asarray(a)
```
If this is needed, I don't think we want `np.array` because that always creates a copy, even if `a` is already an array. In practice, this is only needed because the tests pass in lists - but stats functions always pass in arrays.

Reply:
> but stats functions always pass in arrays.
OK. Let's assume a is an array.